skyportal/models.py

```diff
@@ -459,6 +463,64 @@ def gal_lon_deg(self):
         coord = ap_coord.SkyCoord(self.ra, self.dec, unit="deg")
         return coord.galactic.l.deg
 
+    @property
+    def luminosity_distance(self):
+        """
+        The luminosity distance in Mpc, using either DM or distance data
+        in the altdata fields or using the cosmology/redshift
+
+        Return None if the redshift puts the source not within the Hubble flow
+        """
+
+        # there may be a non-redshift based measurement of distance
+        # for nearby sources
+        if self.altdata:
```

Comment:
Is there a way to incorporate photo-z's here? Or if there are somehow multiple z estimates (no firm spec-z)?

Reply:
A user (currently only via the API) can add `dm`, `parallax`, `dist_kpc`, `dist_Mpc`, `dist_pc` or `dist_cm` to `altdata` and those will be picked up (in that order) as the distance rather than the redshift.

Reply:
Added some more info in the docstrings about this